src/python_testing/TC_MCORE_FS_1_2.py

```diff
@@ -0,0 +1,185 @@
+                 TestStep(6, "TH_FSA initiates commissions of TH_SED_L by sending the OpenCommissioningWindow command to the Administrator Commissioning Cluster on the endpoint with the uniqueID matching that of TH_SED_L."),
+                 TestStep(7, "TH_FSA completes commissioning of TH_SED_L using the Enhanced Commissioning Method.")]
+        return steps
+
+    @async_test_body
+    async def test_TC_MCORE_FS_1_2(self):
+        self.is_ci = self.check_pics('PICS_SDK_CI_ONLY')
+        min_report_interval_sec = self.user_params.get("min_report_interval_sec", 0)
+        max_report_interval_sec = self.user_params.get("max_report_interval_sec", 10 * 60)
+        timeout_delay_sec = self.user_params.get("timeout_delay_sec", max_report_interval_sec * 2)
+
+        self.step(1)
+
+        # Subscribe to the UniqueIDs
+        unique_id_queue = queue.Queue()
+        subscription_contents = [
+            (Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID)  # On all endpoints
+        ]
+        sub = await self.default_controller.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=subscription_contents,
+            reportInterval=(min_report_interval_sec, max_report_interval_sec),
+            keepSubscriptions=False
+        )
+        attribute_handler = AttributeChangeAccumulator(
+            name=self.default_controller.name, expected_attribute=Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID, output=unique_id_queue)
+        sub.SetAttributeUpdateCallback(attribute_handler)
+        sub_handlers.append(attribute_handler)
+
+        logging.info("Waiting for First BridgedDeviceBasicInformation.")
+        start_time = time.time()
+        elapsed = 0
+        time_remaining = timeout_delay_sec
+
+        th_sed_dut_bdbi_endpoint = -1
+        th_sed_dut_unique_id = -1
+
+        while time_remaining > 0 and th_sed_dut_bdbi_endpoint < 0:
+            try:
+                item = unique_id_queue.get(block=True, timeout=time_remaining)
+                client_name, endpoint, attribute, value = item['name'], item['endpoint'], item['attribute'], item['value']
+
+                # Record arrival of an expected subscription change when seen
+                if attribute == Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID:
+                    th_sed_dut_bdbi_endpoint = endpoint
+                    th_sed_dut_unique_id = value
+
+            except queue.Empty:
+                # No error, we update timeouts and keep going
+                pass
+
+            elapsed = time.time() - start_time
+            time_remaining = timeout_delay_sec - elapsed
+
+        logging.info("Found BDBI with UniqueID (%d) on endpoint %d." % th_sed_dut_unique_id, th_sed_dut_bdbi_endpoint)
+
+        self.step(2)
+
+        self.sync_passcode = 20202024
```

Comment:
you can replace all this with openCommissioningWindow from matter_testing support and it's much cleaner. The reason I did it this way in the mock is because that's what the cluster command sends back. If you're just trying to commission this device into a second fabric in the TH, which looks like what's happening here. That'll randomize the discriminator for you and calculate the pake verifier internally so you don't have to hard code it.

follow up here is fine.

Reply:
It doesn't support providing the endpoint number, which is required here.

Reply:
If this is something you're expecting a standard controller to be able to do, it should be added. Follow up is fine.

Reply:
SG, will follow up.